controllers/db/bytebase/controllers/sync.go

```diff
@@ -0,0 +1,288 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/jaevor/go-nanoid"
+	bbv1 "github.com/labring/sealos/controllers/db/bytebase/api/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *BytebaseReconciler) syncIngress(ctx context.Context, bb *bbv1.Bytebase, hostname string) error {
+	var err error
+	host := hostname + DomainSuffix
+	switch bb.Spec.IngressType {
+	case bbv1.Nginx:
+		err = r.syncNginxIngress(ctx, bb, host)
+	//TODO: support apisix
+	default:
+		err = r.syncNginxIngress(ctx, bb, host)
+	}
+	return err
+}
+
+func (r *BytebaseReconciler) syncNginxIngress(ctx context.Context, bb *bbv1.Bytebase, host string) error {
+	ingress := &networkingv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bb.Name,
+			Namespace: bb.Namespace,
+		},
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, ingress, func() error {
+		cookies, err := r.getClientCookie()
+		if err != nil {
+			return err
+		}
+		wholeSnippet := `more_clear_headers "X-Frame-Options:"; more_set_headers "Content-Security-Policy: default-src * blob: data: *.cloud.sealos.io cloud.sealos.io; img-src * data: blob: resource: *.cloud.sealos.io cloud.sealos.io; connect-src * wss: blob: resource:; style-src 'self' 'unsafe-inline' blob: *.cloud.sealos.io cloud.sealos.io resource:; script-src 'self' 'unsafe-inline' 'unsafe-eval' blob: *.cloud.sealos.io cloud.sealos.io resource: *.baidu.com *.bdstatic.com; frame-src 'self' cloud.sealos.io mailto: tel: weixin: mtt: *.baidu.com; frame-ancestors 'self' https://cloud.sealos.io https://*.cloud.sealos.io"; more_set_headers "X-Xss-Protection: 1; mode=block"; %s if ($request_uri ~* \.(js|css|gif|jpe?g|png)) { expires 30d; add_header Cache-Control "public";}`
```

Comment:
All hard coded domain `cloud.sealo.io` or `*.cloud.sealos.io` need to be formatted to real deploy env like `cloud.sealos.cn`

Reply:
This has been resolved in the latest commits